homeassistant/components/xiaomi_miio/sensor.py

```diff
@@ -78,9 +85,20 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
     """Set up the Xiaomi sensor from a config entry."""
     entities = []
 
-    # Gateway sub devices
     if config_entry.data[CONF_FLOW_TYPE] == CONF_GATEWAY:
         gateway = hass.data[DOMAIN][config_entry.entry_id]
+        # Gateway illuminance sensor
+        if gateway.model not in [
+            GATEWAY_MODEL_AC_V1,
+            GATEWAY_MODEL_AC_V2,
+            GATEWAY_MODEL_AC_V3,
+        ]:
```

Comment:
Just for the future, I think that when/if there is a common base class for all gateway-like devices (including those AC ones), I think there should be API to inform the downstream users about the supported features (like this sensor).

Reply:
You mean when not using HomeAssistant but the python-miio library directly?
That sounds like a good idea, but I think there is still some work and re-factoring needed for the python-miio gateway implementation before it makes sence to include such functions...

Reply:
I mean when using it inside homeassistant, the gateway devices (and those AC devices) can have different feature sets. In the long run it would be a good idea to refactor the common functionalities & add a way to check what features are supported by the device at hand. I just wanted to plant the idea there as you are the one who's been touching those parts lately, it's not something that's to be done for this PR.